hack/generate/samples/internal/go/memcached-with-webhooks/memcached_with_webhooks.go

```diff
@@ -787,21 +829,170 @@ import (
+	background = "This document aims to help users that are not familiar with metrics exposed by the memcached-operator.\n" +
+		"All metrics documented here are auto-generated by the utility tool \"monitoring/metricsdocs\" and reflects exactly what is being exposed.\n\n"
+
+	KVSpecificMetrics = "## Memcached Operator Metrics List\n"
+
+	opening = title +
+		background +
+		KVSpecificMetrics
+
+	// footer
+	footerHeading = "## Developing new metrics\n"
+	footerContent = "After developing new metrics or changing old ones, please run \"make generate-metricsdocs\" to regenerate this document.\n\n" +
+		"If you feel that the new metric doesn't follow these rules, please change \"monitoring/metricsdocs\" according to your needs.\n"
+
+	footer = footerHeading + footerContent
+)
+
+// TODO: scaffolding these helpers with operator-lib: https://github.com/operator-framework/operator-lib.
+
+// MemcachedMetricList contains the name, description, and type for each metric.
+func main() {
+	MemcachedMetricList := metricDescriptionListToMetricList(monitoring.ListMetrics())
+	sort.Sort(MemcachedMetricList)
+	writeToFile(MemcachedMetricList)
+}
+
+// writeToFile receives a list of metrics and writes them to a file.
+func writeToFile(metricsList metricList) {
+	fmt.Print(opening)
+	metricsList.writeOut()
+	fmt.Print(footer)
+}
+
+// Metric is an exported struct that defines the metric
+// name, description, and type as a new type named Metric.
+type Metric struct {
+	name        string
+	description string
+	metricType  string
+}
+
+func metricDescriptionToMetric(md monitoring.MetricDescription) Metric {
+	return Metric{
+		name:        md.Name,
+		description: md.Help,
+		metricType:  md.Type,
+	}
+}
+
+// writeOut receives a metric of type metric and prints
+// the metric name, description, and type.
+func (m Metric) writeOut() {
+	fmt.Println("###", m.name)
+	fmt.Println(m.description, "Type: "+m.metricType+".")
+}
+
+// metricList is an array that contain metrics from type metric,
+// as a new type named metricList.
+type metricList []Metric
+
+// metricDescriptionListToMetricList collects the metrics exposed by the
+// memcached-operator, and inserts them into the metricList array.
+func metricDescriptionListToMetricList(mdl []monitoring.MetricDescription) metricList {
+	res := make([]Metric, len(mdl))
+	for i, md := range mdl {
+		res[i] = metricDescriptionToMetric(md)
+	}
+
+	return res
+}
+
+// Len implements sort.Interface.Len
+func (m metricList) Len() int {
+	return len(m)
+}
+
+// Less implements sort.Interface.Less
+func (m metricList) Less(i, j int) bool {
+	return m[i].name < m[j].name
+}
+
+// Swap implements sort.Interface.Swap
+func (m metricList) Swap(i, j int) {
+	m[i], m[j] = m[j], m[i]
+}
+
+func (m metricList) writeOut() {
+	for _, met := range m {
+		met.writeOut()
+	}
+}
+`
+const metricsmdFragment = `# Memcached Operator Metrics
+This document aims to help users that are not familiar with metrics exposed by the memcached-operator.
```

Comment:
by the memcached-operator -> by the operator

Reply:
Agree

Reply:
updated